### ISSUE TITLE
hashCode() on RoaringBitmap

### ISSUE THREAD

**Issue by okrische** [lemire/RoaringBitmap#46]
Hello,

i create 1000 identical RoaringBitmap instances. Then i put them into a set. 

Without runOptimize(), the size is 1 as expected. 
With runOptimize(), the size is 1000, and not 1.

I would assume, runOptimize() should not make any difference. The hashCode is the problem. equals() works fine.

Here are my tests:

```java
	@Test
	public void testRoaringWithOptimize() {
		// create the same bitmap over and over again, with optimizing it
		final Set<RoaringBitmap> setWithOptimize = Sets.newHashSet();
		final int max = 1000;
		for(int i = 0; i < max; i++) {
			final RoaringBitmap bitmapWithOptimize = new RoaringBitmap();
			bitmapWithOptimize.add(1);
			bitmapWithOptimize.add(2);
			bitmapWithOptimize.add(3);
			bitmapWithOptimize.add(4);
			bitmapWithOptimize.runOptimize();
			setWithOptimize.add(bitmapWithOptimize);
		}
		assertEquals(1, setWithOptimize.size());		
	}
	
	@Test
	public void testRoaringWithoutOptimize() {
		// create the same bitmap over and over again, without optimizing it
		final Set<RoaringBitmap> setWithoutOptimize = Sets.newHashSet();
		final int max = 1000;
		for(int i = 0; i < max; i++) {
			final RoaringBitmap bitmapWithoutOptimize = new RoaringBitmap();
			bitmapWithoutOptimize.add(1);
			bitmapWithoutOptimize.add(2);
			bitmapWithoutOptimize.add(3);
			bitmapWithoutOptimize.add(4);
			setWithoutOptimize.add(bitmapWithoutOptimize);
		}
		assertEquals(1, setWithoutOptimize.size());
	}
```

**Comment by lemire** [comment]
@okrische 

Good one. Yes. RunContainers do not implement hashCode... hence the problem.

**Comment by lemire** [comment]
@okrische 

The fix is easy. Need to release.

**Comment by lemire** [comment]
Fixed in RoaringBitmap-0.5.5.

Thanks for the bug report.
